### DISCUSSION TITLE
Telnet host mode - transparent to the host?

### DISCUSSION THREAD

**Discussion by pheller** [bozimmerman/Zimodem#148] [General]
Hi,

I'm trying to use the Retromodem (which uses a derivative of the Zimodem, as I understand) with an old DOS BBS package.  No matter what combination of S41 and ATA/ATAT I use, it seems that an inbound telnet connection will case something like:

```
CONNECT 9600
[ 37 15 245 ]
<a few binary bytes>
```

I presume this is translation of telnet options from binary to a well formed ASCII representation, and the binary bits are probably parts of the TELNET options that were not processed.

Is this expected behavior?  Is there a configuration to make this completely transparent to the host without presenting this information?

Thanks!

**Comment by joeventura1** [comment]
Sadly I am experiencing the same issue.

**Reply by sswazey** [reply]
I think this is just a config problem.  You must be in BBS mode,  telnet on, and BBS echo off (not keystrokes echo).  Assuming you're using the retromodem, make sure you have the latest version by typing:

**AT&U=3.7.3**

This will force and upgrade even if you are already at 3.7.3

Type the following **BOLD** text when prompted:

Zimodem Firmware v3.7.3-retro-audio
sdk=2.2.2-dev(38a443e) chipid=1458270 cpu@80
totsize=4096k ssize=742k fsize=1907k speed=40m
CONNECTED TO WiFi (192.168.5.61)
READY.


**AT+CONFIG**

Main Menu
[HOST] name: retromodem-1
[WIFI] connection: WiFi
[FLOW] control: DISABLED
[ECHO] keystrokes: ON
[BBS] host: DISABLED
[PETSCII] translation: OFF
[ADD] new phonebook entry

Enter command or entry or ENTER to exit: **BBS**

BBS host settings:

[ENABLE] BBS host listener

Enter option to toggle or ENTER to exit
: **ENABLE**

BBS host settings:

[HOST] Listener Port: 6502
[PETSCII] Translation: OFF
[TELNET] Translation: OFF
[ECHO]: ON
[FLOW] Control: DISABLED
[DISABLE] BBS host listener

Enter option to toggle or ENTER to exit
: **TELNET**

BBS host settings:

[HOST] Listener Port: 6502
[PETSCII] Translation: OFF
[TELNET] Translation: ON
[ECHO]: ON
[FLOW] Control: DISABLED
[DISABLE] BBS host listener

Enter option to toggle or ENTER to exit
: **ECHO**

BBS host settings:

[HOST] Listener Port: 6502
[PETSCII] Translation: OFF
[TELNET] Translation: ON
[ECHO]: OFF
[FLOW] Control: DISABLED
[DISABLE] BBS host listener

Enter option to toggle or ENTER to exit
: **HOST**

Enter a port number to listen on
: **23**

BBS host settings:

[HOST] Listener Port: 23
[PETSCII] Translation: OFF
[TELNET] Translation: ON
[ECHO]: OFF
[FLOW] Control: DISABLED
[DISABLE] BBS host listener

Enter option to toggle or ENTER to exit
: **\<ENTER\>**

Main Menu
[HOST] name: retromodem-1
[WIFI] connection: WiFi
[FLOW] control: DISABLED
[ECHO] keystrokes: ON
[BBS] host: Port 23
[PETSCII] translation: OFF
[ADD] new phonebook entry

Enter command or entry or ENTER to exit: **\<ENTER\>**

Your setting changed. Save (y/N)?  **Y**

Settings saved.

Zimodem Firmware v3.7.3-retro-audio
sdk=2.2.2-dev(38a443e) chipid=1458270 cpu@80
totsize=4096k ssize=742k fsize=1907k speed=40m
CONNECTED TO WiFi (192.168.5.61)
READY.

Now setup the modem for incoming calls:

**ATS0=1**
**ATS41=1**
**AT&W**

The modem should illuminate the AA LED.

Use telnet to connect to the device. In my setup, the modem is at 192.168.5.61 so
ubuntu%  **telnet 192.168.5.61**

The modem, on the serial port side,  should respond:
RING
CONNECT 1200

**Reply by pheller** [reply]
The config above does appear to fully resolve the issue with the telnet translation.  I swear I tried this exact config before.  My retromodem was definitely on a previous firmware, so maybe a combo of config and whatever changes went into 3.7.3.

Regardless, it works great now.  Thanks!

**Comment by sswazey** [comment]
When using the zimodem with a bbs as the dial-in modem you should also set the following:

AT E0  ; no echo in AT command mode
AT Q1 ; Quiet, no AT response text
AT &W ; save settings

or 
ATS0=1S41=1E0Q1&W

**Comment by joeventura1** [comment]
Thank you!!!
